src/widgets/RowDetail.js

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import React from 'react';
+import { View, TouchableOpacity, StyleSheet, Keyboard } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Modal from 'react-native-modalbox';
+
+import { ItemDetails } from './bottomModals';
+import { CloseIcon } from './icons';
+import { RowDetailActions, ROW_DETAIL_KEYS } from '../actions/RowDetailActions';
+import { RequisitionItemDetails } from './RequisitionItemDetail';
+
+import { DARKER_GREY } from '../globalStyles';
+import { selectRowDetailIsOpen } from '../selectors/rowDetail';
+
+/**
+ * Container component for a presentation component that will display
+ * details for a row.
+ */
+export const RowDetailComponent = ({
+  detailKey,
+  rowData,
+  isOpen,
+  onClose,
+  swipeToClose,
+  backdropPressToClose,
+  position,
+  backdrop,
+  modalStyle,
+  headerRowStyle,
+  containerStyle,
+  dismissKeyboardOnOpen,
+  dismissOnKeyboardOpen,
+}) => {
+  // This component renders at the bottom of the screen. Dismiss the keyboard when rendering
+  // so the full screen isn't covered by keyboard and this modal.
+  React.useEffect(() => {
+    if (isOpen && dismissKeyboardOnOpen) Keyboard.dismiss();
+  }, [isOpen]);
+
+  // Add/removing a listener on mounting/unmounting which will hide this component when the
+  // keyboard shows.
+  React.useEffect(() => {
+    if (dismissOnKeyboardOpen) {
+      Keyboard.addListener('keyboardDidShow', onClose);
+      return () => Keyboard.removeListener('keyboardDidShow', onClose);
+    }
+    return null;
+  }, []);
+
+  const getDetailComponent = () => {
+    switch (detailKey) {
+      case ROW_DETAIL_KEYS.ITEM_DETAIL:
+        return <ItemDetails item={rowData} />;
+      case ROW_DETAIL_KEYS.REQUISITION_ITEM_DETAIL:
+        return <RequisitionItemDetails item={rowData} />;
+      default:
+        return null;
```

Comment:
so plan would be to just render a different component for different rows. Seems nicer than an even more generic solution